app/uk/gov/hmrc/helptosavereminder/repo/HtsReminderRepository.scala

```diff
@@ -104,20 +105,32 @@ class HtsReminderMongoRepository @Inject()(mongo: ReactiveMongoComponent)
 
   }
 
-  override def updateEmail(nino: String, firstName: String, lastName: String, email: String): Future[Boolean] = {
+  override def updateEmail(nino: String, firstName: String, lastName: String, email: String): Future[Int] = {
     val selector = Json.obj("nino" -> nino)
     val modifier = Json.obj("$set" -> Json.obj("email" -> email, "firstName" -> firstName, "lastName" -> lastName))
     val result = proxyCollection.update(ordered = false).one(selector, modifier)
 
     result
       .map { status =>
         Logger.debug(s"[HtsReminderMongoRepository][updateEmail] updated:, result : $status ")
-        statusCheck("Failed to update HtsUser Email, No Matches Found", status)
+
+        status.n match {
+          case 0 => {
+            Logger.error("Failed to update HtsUser Email, No Matches Found")
+            NOT_FOUND
+          }
+          case _ =>
+            status.nModified match {
```

Comment:
Better not to nest matches, but instead match on (status.n, status.nModified)

Reply:
Done